src/Compilers/Core/Portable/Diagnostic/DiagnosticDescriptor.cs

```diff
@@ -170,15 +170,17 @@ public sealed class DiagnosticDescriptor : IEquatable<DiagnosticDescriptor>
         public bool Equals(DiagnosticDescriptor other)
         {
             return
-                other != null &&
-                this.Category == other.Category &&
-                this.DefaultSeverity == other.DefaultSeverity &&
-                this.Description.Equals(other.Description) &&
-                this.HelpLinkUri == other.HelpLinkUri &&
-                this.Id == other.Id &&
-                this.IsEnabledByDefault == other.IsEnabledByDefault &&
-                this.MessageFormat.Equals(other.MessageFormat) &&
-                this.Title.Equals(other.Title);
+                other is
+            {
+                } && this is
+            {
+                Category: other.Category,
```

Comment:
>other.Category [](start = 26, length = 14)

Not a valid pattern.

Reply:
➡️ Treating this one as a duplicate of https://github.com/dotnet/roslyn/pull/39738#discussion_r344416577